gtk4/src/subclass/widget.rs

```diff
@@ -172,27 +167,18 @@ pub trait WidgetImplExt: ObjectSubclass {
 }
 
 impl<T: WidgetImpl> WidgetImplExt for T {
-    fn parent_compute_expand(
-        &self,
-        widget: &Self::Type,
-        hexpand_p: &mut bool,
-        vexpand_p: &mut bool,
-    ) {
+    fn parent_compute_expand(&self, widget: &Self::Type, hexpand: &mut bool, vexpand: &mut bool) {
         unsafe {
             let data = T::type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkWidgetClass;
-            let f = (*parent_class)
-                .compute_expand
-                .expect("No parent class impl for \"compute_expand\"");
-            let mut h: i32 = hexpand_p.to_glib();
-            let mut v: i32 = vexpand_p.to_glib();
-            f(
-                widget.unsafe_cast_ref::<Widget>().to_glib_none().0,
-                &mut h,
-                &mut v,
-            );
-            *hexpand_p = from_glib(h);
-            *vexpand_p = from_glib(v);
+            let widget = widget.unsafe_cast_ref::<Widget>();
+            if let Some(f) = (*parent_class).compute_expand {
+                f(
+                    widget.to_glib_none().0,
+                    &mut hexpand.to_glib(),
+                    &mut vexpand.to_glib(),
```

Comment:
Ah they're bools, for some reason I thought they're i32s.

Then you need to

```rust
let hexpand_glib = hexpand.to_glib();
[...]
f(..., &mut hexpand_glib, ...);
*hexpand = hexpand_glib;
[...]
```

Reply:
Looks good apart from that now. Can you also send the same PR to gtk3?

Reply:
Otherwise the problem is that you create a temporary local variable, pass that to C, C modifies that, and then it's lost.